samza-core/src/main/java/org/apache/samza/clustermanager/StandbyContainerManager.java

```diff
@@ -181,13 +207,14 @@ private void handleStandbyContainerStop(String standbyContainerID, String resour
 
       // request standbycontainer's host for active-container
       SamzaResourceRequest resourceRequestForActive =
-          containerAllocator.getResourceRequestWithDelay(activeContainerID, standbyContainerHostname, preferredHostRetryDelay);
+        containerAllocator.getResourceRequestWithDelay(activeContainerID, standbyContainerHostname, preferredHostRetryDelay);
       // record the resource request, before issuing it to avoid race with allocation-thread
       failoverMetadata.get().recordResourceRequest(resourceRequestForActive);
       containerAllocator.issueResourceRequest(resourceRequestForActive);
 
       // request any-host for standby container
-      containerAllocator.requestResource(standbyContainerID, ResourceRequestState.ANY_HOST);
+      containerAllocator.requestResource(standbyContainerID, ResourceRequestState.ANY_HOST,
```

Comment:
I thought about this a lot and I think it makes sense to not alter the current flow at all in case the config is turned on. Hence I have now added that check for every `requestResource`.